modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/capability/xml/schema/AbstractXmlResourceFactory.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.capability.xml.schema;
+
+import static org.mule.runtime.module.extension.internal.capability.xml.schema.model.SchemaConstants.XSD_EXTENSION;
+import org.mule.runtime.extension.api.introspection.ExtensionModel;
+import org.mule.runtime.extension.api.introspection.property.XmlModelProperty;
+import org.mule.runtime.extension.api.resources.GeneratedResource;
+import org.mule.runtime.extension.api.resources.spi.GeneratedResourceFactory;
+
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * Base class for {@link GeneratedResourceFactory} implementations which
+ * generate resources needed to support configuring extensions through
+ * XML
+ *
+ * @since 4.0
+ */
+abstract class AbstractXmlResourceFactory implements GeneratedResourceFactory
+{
+
+    /**
+     * Tests the given {@code extensionModel} to be enriched with the {@link XmlModelProperty}.
```

Comment:
It's kind of the idea since this is an abstract and protected class. The reader should know how it works so that he knows what will happen when extending it